### PR TITLE
NetBSD support proposal.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by Rumi-Larry** [comment]
If someone with a pc running NetBSD can test if it builds and runs, then I don't see great harm in adding this. Particularly considering there's Haiku support.

File: Source/Core/Common/Thread.cpp [hunk 1]
```diff
@@ -119,19 +121,30 @@ void SetThreadAffinity(std::thread::native_handle_type thread, u32 mask)
...
 #else
   cpu_set_t cpu_set;
 #endif
+#ifndef __NetBSD__
   CPU_ZERO(&cpu_set);
+#endif
 
   for (int i = 0; i != sizeof(mask) * 8; ++i)
     if ((mask >> i) & 1)
+#ifndef __NetBSD__
```
Comment (leoetlino) [inline Source/Core/Common/Thread.cpp:138]: Please make two ifndef blocks. This looks horribly error prone. The difference in indentation looked like a style issue at first, until I realised the cpuset_set was guarded by an if statement...

File: Source/Core/Common/StringUtil.cpp [hunk 2]
```diff
@@ -524,8 +525,11 @@ std::string CodeTo(const char* tocode, const char* fromcode, std::basic_string_v
     while (src_bytes != 0)
     {
       size_t const iconv_result =
+#if defined(__OpenBSD__) || defined(__NetBSD__)
+          iconv(conv_desc, (const char**)&src_buffer, &src_bytes, &dst_buffer, &dst_bytes);
```
Comment (leoetlino) [inline Source/Core/Common/StringUtil.cpp:529]:
```suggestion
          iconv(conv_desc, reinterpret_cast<const char**>(&src_buffer), &src_bytes, &dst_buffer, &dst_bytes);
```
(you might need to re-run clang-format after applying this suggestion)

**Review by leoetlino — CHANGES_REQUESTED** [review]

**Comment by leoetlino** [comment]
@devnexen This fails to build on most builders. You may need a const_cast to fix the issue.

File: Source/Core/Common/StringUtil.cpp [hunk 3]
```diff
@@ -524,8 +526,13 @@ std::string CodeTo(const char* tocode, const char* fromcode, std::basic_string_v
     while (src_bytes != 0)
     {
       size_t const iconv_result =
-          iconv(conv_desc, (char**)(&src_buffer), &src_bytes, &dst_buffer, &dst_bytes);
-
+#if defined(__OpenBSD__) || defined(__NetBSD__)
+          iconv(conv_desc, reinterpret_cast<const char**>(&src_buffer), &src_bytes, &dst_buffer,
+                &dst_bytes);
+#else
+          iconv(conv_desc, const_cast<char**>(reinterpret_cast<char**>(&src_buffer)), &src_bytes, &dst_buffer,
```
Comment (leoetlino) [inline Source/Core/Common/StringUtil.cpp:533]:
```suggestion
          iconv(conv_desc, const_cast<char**>(reinterpret_cast<const char**>(&src_buffer)), &src_bytes, &dst_buffer,
```

**Review by leoetlino — CHANGES_REQUESTED** [review]

**Review by leoetlino — APPROVED** [review]

**Pull request merged**